src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -240,14 +253,31 @@ public enum MysteriousTreasures implements LogicCardInfo {
           pokePower "Power Cancel", {
             text "Once during your opponent’s turn, when your opponent’s Pokémon uses any Poké-Power, you may discard 2 cards from your hand and prevent all effects of that Poké-Power. (This counts as that Pokémon using its Poké-Power.) This power can’t be used if Alakazam is affected by a Special Condition."
             actionA {
+              //TODO: Yeah this is gonna be a fun one to do.
+              //Requires a way to capture "powerUsed", stop the Poké-Power at that point but make it count as used.
             }
           }
           move "Psychic Guard", {
-            text "50 damage. ."
+            text "50 damage. During your opponent’s next turn, any damage done to Alakazam by attacks from your opponent’s Stage 2 Pokémon is reduced by 30 (after applying Weakness and Resistance)."
             energyCost P, P, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 50
+              afterDamage {
+                delayed {
+                  before APPLY_ATTACK_DAMAGES, {
+                    bg.dm().each {
+                      if (it.to==self && it.dmg.value && it.notNoEffect && it.from.topPokemonCard.cardTypes.is(STAGE2)) {
+                        bc "-30 to Alakazam (Psychic Guard)"
```

Comment:
Should probably use `$self` here, or not mention the Pokémon at all? I assume if the attack is copied then it would display this exact text still.